packages/application-shell/src/components/with-applications-menu/with-applications-menu.js

```diff
@@ -68,20 +68,17 @@ function withApplicationsMenu(getOptions) {
             >
               {({ data, error }) => (
                 <Component
-                  {...props}
-                  {...{ [queryName]: { ...data, error } }}
+                  {...this.props}
+                  {...{ [mergedConfig.queryName]: { ...data, error } }}
                 />
               )}
             </Query>
           )}
         />
       );
-    };
-    WrappedComponent.displayName = `withApplicationsMenu(${getDisplayName(
```

Comment:
Ah, this just moved up.

Reply:
It's now one single component within the HOC, with some if-else.